src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -233,7 +233,7 @@ public HubConnection(IConnectionFactory connectionFactory, IHubProtocol protocol
         /// </summary>
         /// <param name="cancellationToken">The token to monitor for cancellation requests. The default value is <see cref="CancellationToken.None" />.</param>
         /// <returns>A <see cref="Task"/> that represents the asynchronous start.</returns>
-        public async Task StartAsync(CancellationToken cancellationToken = default)
+        public virtual async Task StartAsync(CancellationToken cancellationToken = default)
         {
             CheckDisposed();
             using (_logger.BeginScope(_logScope))
```

Comment:
The logs in the override method won't have the log scope. Is it important?

How about adding new virtual methods which are already wrapped in the log scope?

```
public async Task StartAsync(CancellationToken cancellationToken = default)
{  
    using (_logger.BeginScope(_logScope))
    {
        await NewMethodStartAsync(cancellationToken);
    }
}

public virtual async Task NewMethodStartAsync(CancellationToken cancellationToken = default)
{
    CheckDisposed();
    await StartAsyncInner(cancellationToken).ForceAsync();
}
```

Reply:
I think this is to much ceremony for what we're trying to achieve. This is about testability, not about being able to put custom code into our logging scopes. If you want to create an outer scope for your override, you can still do that.